cdap-client/src/main/java/co/cask/cdap/client/ProgramClient.java

```diff
@@ -356,10 +357,10 @@ public String getStatus(ProgramId programId) throws IOException, ProgramNotFound
    * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
    * @throws TimeoutException if the program did not achieve the desired program status before the timeout
    * @throws InterruptedException if interrupted while waiting for the desired program status
-   * @deprecated since 4.0.0. Please use {@link #waitForStatus(ProgramId, String, long, TimeUnit)} instead
+   * @deprecated since 4.0.0. Please use {@link #waitForStatus(ProgramId, ProgramStatus, long, TimeUnit)} instead
    */
   @Deprecated
-  public void waitForStatus(final Id.Program program, String status, long timeout, TimeUnit timeoutUnit)
+  public void waitForStatus(final Id.Program program, ProgramStatus status, long timeout, TimeUnit timeoutUnit)
```

Comment:
shouldn't change this, since it is incompatible. Instead, this should do ``waitForStatus(program.toEntityId(), ProgramStatus.valueOf(status), timeout, timeoutUnit);``